cocotb/__init__.py

```diff
@@ -106,8 +106,67 @@ def _reopen_stream_with_buffering(stream_name):
 plusargs = {}
 """A dictionary of "plusargs" handed to the simulation."""
 
-# To save typing provide an alias to scheduler.add
-fork = scheduler.add
+
+def run_task(coro):
+    """
+    Runs a coroutine concurrently. Can be awaited at any point to block until a result is available.
+
+    .. versionadded:: 1.4
+    """
+    return scheduler.add(coro)
+
+
+def fire_and_forget(coro):
+    """
+    Runs a coroutine concurrently that is never meant to be awaited. Will end the test if an exception occurs.
+
+    .. versionadded:: 1.4
+    """
+    co = cocotb.decorators.make_task(coro)
+
+    async def watcher():
+        try:
+            res = await co
+        except Exception as e:
+            cocotb.utils.remove_traceback_frames(e, ['watcher', '__await__'])
+            log.warning("Ending test, fire and forget coroutine {} resulted in an exception.".format(coro))
+            scheduler.finish_test(e)
+        else:
+            log.info("Fire and forget coroutine {} resulted in {}.".format(coro, res))
+
+    return run_task(watcher())
+
+
+def fork(coro):
+    """
+    Runs a coroutine concurrently. Will end the current test *if* the coroutine results in an exception *before* being
```

Comment:
```suggestion
    Run a coroutine concurrently. 
    
    This will end the current test *if* the coroutine results in an exception *before* being
```

Reply:
Was this an accidental "resolve"?